server/src/main/java/org/opensearch/rest/extensions/RestSendToExtensionAction.java

```diff
@@ -86,15 +91,26 @@ public RestSendToExtensionAction(
 
         List<Route> restActionsAsRoutes = new ArrayList<>();
         for (String restAction : restActionsRequest.getRestActions()) {
-            int delim = restAction.indexOf(' ');
+            Optional<String> name = Optional.empty();
             try {
-                method = RestRequest.Method.valueOf(restAction.substring(0, delim));
-                path = pathPrefix + restAction.substring(delim).trim();
+                String[] parts = restAction.split(" ");
+                if (parts.length < 2) {
+                    throw new IllegalArgumentException(restAction + " does not begin with a valid REST method");
+                }
```

Comment:
This check can be removed since Method.valueOf fails with IllegalArgumentException  and is being caught in catch block below?

Reply:
Moved this up and changed the message to match the error.